src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -263,11 +266,12 @@ private static HttpResponse handleExceptionalStatus(
           requestBuilder.addToken(request);
           response = httpClient.execute(request);
           if (!isStatusOK(response.getStatusLine())) {
```

Comment:
Is the if-condition needed here? It looks like we are already inside the same condition (line 247).

Reply:
ah, see line 262-265, we refresh the token and retry the request again